cmd/jujud/agent/caasunitinit.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package agent
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/loggo"
+	"gopkg.in/juju/names.v3"
+
+	"github.com/juju/juju/agent/tools"
+	"github.com/juju/juju/caas"
+	jujucmd "github.com/juju/juju/cmd"
+	cmdutil "github.com/juju/juju/cmd/jujud/util"
+	"github.com/juju/juju/juju/sockets"
+	"github.com/juju/juju/worker/uniter"
+	"github.com/juju/juju/worker/uniter/runner/jujuc"
+)
+
+// CAASUnitInitCommand represents a jujud bootstrap command.
+type CAASUnitInitCommand struct {
+	cmd.CommandBase
+
+	Wait bool
+	Send bool
+	Args struct {
+		Unit               string `json:"unit"`
+		OperatorFile       string `json:"operator-file"`
+		OperatorCACertFile string `json:"operator-ca-cert-file"`
+		CharmDir           string `json:"charm-dir"`
+	}
+
+	copyFunc      func(string, string) error
+	symlinkFunc   func(string, string) error
+	removeAllFunc func(string) error
+	mkdirAllFunc  func(string, os.FileMode) error
+}
+
+// NewCAASUnitInitCommand returns a new CAASUnitInitCommand that has been initialized.
+func NewCAASUnitInitCommand() *CAASUnitInitCommand {
+	return &CAASUnitInitCommand{
+		copyFunc:      copy,
+		symlinkFunc:   os.Symlink,
+		removeAllFunc: os.RemoveAll,
+		mkdirAllFunc:  os.MkdirAll,
+	}
+}
+
+// Info returns a description of the command.
+func (c *CAASUnitInitCommand) Info() *cmd.Info {
+	return jujucmd.Info(&cmd.Info{
+		Name:    "caas-unit-init",
+		Purpose: "initialize caas unit filesystem",
+	})
+}
+
+func (c *CAASUnitInitCommand) SetFlags(f *gnuflag.FlagSet) {
+	f.BoolVar(&c.Wait, "wait", false, "wait for args init via socket")
+	f.BoolVar(&c.Send, "send", false, "send args for init via socket")
+	f.StringVar(&c.Args.Unit, "unit", "", "unit name")
+	f.StringVar(&c.Args.OperatorFile, "operator-file", "", "operator client info file")
+	f.StringVar(&c.Args.OperatorCACertFile, "operator-ca-cert-file", "", "ca cert for operator")
+	f.StringVar(&c.Args.CharmDir, "charm-dir", "", "directory containing the charm")
+}
+
+func (c *CAASUnitInitCommand) Init(args []string) error {
+	if c.Wait && c.Send {
```

Comment:
what happens if neither are set.
we need at least one right?

Reply:
an answer if resolving?

Reply:
Ahh sorry I added a comment inside the Run method.
// The waiting process waits for arguments to be sent by the
// sending process. If neither wait or send is specified,
// continue with the arguments already specified.